contracts/token/ERC20/extensions/ERC4626.sol

```diff
@@ -179,44 +179,14 @@ abstract contract ERC4626 is ERC20, IERC4626 {
      * would represent an infinite amount of shares.
      */
     function _convertToShares(uint256 assets, Math.Rounding rounding) internal view virtual returns (uint256) {
-        uint256 supply = totalSupply();
-        return
-            (assets == 0 || supply == 0)
-                ? _initialConvertToShares(assets, rounding)
-                : assets.mulDiv(supply, totalAssets(), rounding);
-    }
-
-    /**
-     * @dev Internal conversion function (from assets to shares) to apply when the vault is empty.
-     *
-     * NOTE: Make sure to keep this function consistent with {_initialConvertToAssets} when overriding it.
-     */
-    function _initialConvertToShares(
-        uint256 assets,
-        Math.Rounding /*rounding*/
-    ) internal view virtual returns (uint256 shares) {
-        return assets;
+        return assets.mulDiv(totalSupply() + 10 ** _decimalsOffset(), totalAssets() + 1, rounding);
```

Comment:
does anything weird happen if an attacker mints 1 wei of assets worth of shares, then redeems 10**decimalsOffset - 1 shares to bring the exchange rate back to 1 wei : 1 wei?

Seems like there could still be some manipulation issues

Reply:
It the vault is empty and someone mints 1 wei of assets, it will get `10**offset` "wei of shares".

At the point the vault contains:
- 1 wei of asset + 1 virtual wei of asset
- `10**offset` shares unit + `10**offset` shares unit.

It the user redeems  `10**offset - 1`, we will get 0 asset for it, and the vault will contain:
- 1 wei of asset + 1 virtual wei of asset
- `1` shares unit + `10**offset` shares unit.
So the exchange rate will be `2:10**offset+1` ... very far from `1:1`